CONTRIBUTING.md

```diff
@@ -55,6 +83,260 @@ In addition to the [RuboCop] styles configured described above, we follow common
+      @email  = email
+      @phone  = phone
+    end
+  end
+  ```
+
+### Classes and Modules
+
+* Set `attrs` first
+* Set `class << self` after `attrs`
+* Set  `initialize` after `class << self`
+
+  ```ruby
+  class MyClass
+    attr_reader :something, :another
+
+    class << self
+      def singleton_method
+        ...
+      end
+    end
+
+    def initialize
+      ...
+    end
+  end
+  ```
+
+* Indent `private` and keep `defs` aligned with the block
+
+  ```ruby
+  class MyClass
+    # public methods
+    def public_method
+      ...
+    end
+    
+      private
+
+      def my_private_method
+        ...
+      end
+  end
+  ```
+
+### Collections
+
+* Prefer literal array syntax over `%w` or `%i`
+
+  ```ruby
+  # fail
+  arr_1 = %w(one two three)
+
+  # pass
+  arr_1 = ["one", "two", "three"]
+  ```
+
+* Instantiate with literals for all collections if possible.
+
+  ```ruby
+  # fail
+  arr = Array.new()
+  hsh = Hash.new()
+
+  # pass
+  arr = []
+  hsh = {}
+  ```
+
+### Comments
+
+* We encourage comments in the code.
+* Use proper grammar and punctuation.
+* Focus on *why* the code is the way it is if it is not obvious.
+* Focus less on *how* the code works, that should be evident from the code.
+
+### Exceptions
+
+* Use implicit `begin` in method
+
+  ```ruby
+  # fail
+  def some_method
+    begin
+      ...
+    rescue Psych::SyntaxError => e
+      ...
+      raise e
+  end
+
+  # pass
+  def some_method
+    ...
+  rescue Psych::SyntaxError => e
+    ...
+    raise e
+  end
+
+## Naming
+
+### Variables
+
+* Use meaningful names:
+  * Ruby is not a statically typed language so we need good naming for maintainability.
+  * Consider using an objects type for the dummy variable if possible.
+  * Otherwise, try to convey what the object being passed to the block is through the name:
+
+    ```ruby
+    arr_1 = ['one', 'two', 'three']
+    arr_2 = [1, 2, 3]
+
+    # fail
+    arr_1.each { |e| puts e }
+    arr_2.each { |e| puts e }
+
+    # pass
+    arr_1.each { |str| puts str }
+    arr_2.each { |int| puts int }
+    ```
+
+### Methods
+
+* Avoid `is_` in method names.
+* Use `?` suffix for methods that return a `bool`.
+* Use `save` for boolean return and `save!` with exception returns.
+* Favor functional methods
+  * break up long logic or data transformations into their own methods
+* DRY out code as best you can
+  
+    ```ruby
+    # fail
+    def some_method
+      if var1 && bool2 && x > y || big % small > 1
+        ...
+      end
+    end
+
+    def some_other_method
+      if var1 && bool2 && x > y || big % small > 1
+        ...
+      end
+    end
+
+    # pass
+    def some_method
+      if conditions_true?
+        ...
+      end
+    end
+
+    def some_other_method
+      if conditions_true?
+        ...
+      end
+    end
+
+    def conditions_true?
+      if var1 && bool2 && x > y || big % small > 1
+    end
+    ```
+
+### Classe and Modules
+
+* Camel case for Class and Module names.
```

Comment:
Just to make sure we are clear.  Some people view camel case as 
class camelCase  
some view camel case
class CamelCase

Which is your intent here?
I do understand that the 2nd example is Mountain Case.  However, we should probably make it very clear.

Reply:
Thanks for that! I should have had Mountain/Pascal there. Changes made with example added.